mqtt-client/main.py

```diff
@@ -174,107 +513,235 @@ def input_status(self, status):
+                        thread = Thread(
+                            target=self._send,
+                            args=(self._output_topic.format(id=output_id), data, self._output_qos, self._output_retain)
+                        )
                         thread.start()
             except Exception as ex:
                 self.logger('Error processing outputs: {0}'.format(ex))
 
     @receive_events
-    def recv_events(self, id):
-        if self._enabled is True:
+    def receive_events(self, event_id):
+        if self._enabled and self._event_enabled:
             try:
-                self.logger('Got event {0}'.format(id))
-                data = {'id': id,
-                        'timestamp': time.time()}
-                thread = Thread(target=self._send, args=('openmotics/events/event/{0}'.format(id), data))
+                self._log('Got event {0}'.format(event_id))
+                self.logger('Got event {0}'.format(event_id))
+                data = {'id': event_id,
+                        'timestamp': self._timestamp2isoformat()}
+                thread = Thread(
+                    target=self._send,
+                    args=(self._event_topic.format(id=event_id), data, self._event_qos, self._event_retain)
+                )
                 thread.start()
             except Exception as ex:
                 self.logger('Error processing event: {0}'.format(ex))
 
+    @background_task
+    def background_task_temperature_status(self):
+        self._create_background_task(
+            'temperature',
+            self.webinterface.get_sensor_temperature_status,
+            self._process_sensor_status
+        )()
+
+    @background_task
+    def background_task_humidity_status(self):
+        self._create_background_task(
+            'humidity',
+            self.webinterface.get_sensor_humidity_status,
+            self._process_sensor_status
+        )()
+
+    @background_task
+    def background_task_brightness_status(self):
+        self._create_background_task(
+            'brightness',
+            self.webinterface.get_sensor_brightness_status,
+            self._process_sensor_status
+        )()
+
+    @background_task
+    def background_task_realtime_power(self):
+        self._create_background_task(
+            'power',
+            self.webinterface.get_realtime_power,
+            self._process_realtime_power
+        )()
+
+    @background_task
+    def background_task_total_energy(self):
+        self._create_background_task(
+            'energy',
+            self.webinterface.get_total_energy,
+            self._process_total_energy
+        )()
+
+    def _process_sensor_status(self, sensor_config, json_data):
+        mqtt_messages = []
+        data_list = list(filter(None, json_data.get('status', [])))
+        for sensor_id, sensor_value in enumerate(data_list):
+            sensor_data = {'id': sensor_id,
+                           'name': self._sensors.get(sensor_id).get('name'),
```

Comment:
You use `foo.get(abc).get(xyz)` or similar every now and then. I think this causes the `Error processing power sensor status: 'NoneType' object has no attribute 'get'` I was seeing on my setup. If `abc` is not in `foo`, it will return `None` and the second get will raise this exception. I think it would better to either have one of these:
* If you're sure that `abc` will be in `foo`, then use e.g. `foo[abc].get(xyz, <some fallback>)`.
* Otherwise, do somthing like:

```
config = foo.get(abc)
if config:
    .... config.get(xyz, <fallback value>)
    ...
```

Reply:
Change made in [this commit](https://github.com/DaGolle/openmotics-plugins/commit/8b1c87b50b7f08ef76a283658407325861e4f119).
Did not remove all the .get(...).get(...) occurrences, as some are referencing the configuration and should always work I think.
Thanks for the tips! Hope it will make the plugin more stable.